statistical-model/generate_bell_curves.py

```diff
@@ -0,0 +1,191 @@
+import numpy as np
+import matplotlib.pyplot as plt
+import math
+from scipy.stats import beta
+import json
+
+NUM_CLIPS = 0
+
+def calculate_angle(vertex, a, b):
+    """
+    calculate angle (in degrees) for (a -> vertex -> b)
+
+    Inputs:
+        a       = point 1
+        vertex  = point 2
+        b       = point 3
+
+    Outputs:
+        degrees = degrees between a -> vertex -> b
+    """
+    AV = (a[0] - vertex[0], a[1] - vertex[1])
+    BV = (b[0] - vertex[0], b[1] - vertex[1])
+    
+    dot_product = AV[0]*BV[0] + AV[1]*BV[1]
+    AV_magnitude = math.sqrt(AV[0]**2 + AV[1]**2)
+    BV_magnitude = math.sqrt(BV[0]**2 + BV[1]**2)
+    
+    if AV_magnitude == 0 or BV_magnitude == 0:
+        return 0.0
+    
+    cos_value = max(min(dot_product / (AV_magnitude * BV_magnitude), 1.0), -1.0)
+    return math.degrees(math.acos(cos_value))
+
+def process_arm_data(shoulder, elbow, wrist, pinky, index):
+    """
+    Processes frames of data for one arm and computes two arrays:
+      - sew: Shoulder -> Elbow -> Wrist angles (computed at the elbow)
+      - ewa: Elbow -> Wrist -> Average(Pinky, Index) angles (computed at the wrist)
+    
+    Inputs: 
+        shoulder    = shoulder coordinate 
+        elbow       = elbow coordinate
+        wrist       = wrist coordinate
+        pinky       = pinky coordinate
+        index       = index coordinate
+    
+    Outputs:
+        angle_sew_list  = list of angles for each dataframe in input-file for shoulder->elbow->wrist
+        angle_ewa_list  = list of angles for each dataframe in input-file for elbow->wrist->fingers
+    """
+    angle_sew_list = []  
+    angle_ewa_list = []  
+    
+    for i in range(len(shoulder)):
+        s = shoulder[i]
+        e = elbow[i]
+        w = wrist[i]
+        p = pinky[i]
+        idx = index[i]
+        
+        angle_sew = calculate_angle(e, s, w)
+        angle_sew_list.append(angle_sew)
+        
+        avg_pinky_index = [(p[0] + idx[0]) / 2, (p[1] + idx[1]) / 2]
+        angle_ewa = calculate_angle(w, e, avg_pinky_index)
+        angle_ewa_list.append(angle_ewa)
+    
+    return np.array(angle_sew_list), np.array(angle_ewa_list)
+
+def plot_and_save(angles_sew, angles_ewa, arm_name):
```

Comment:
Can you leave this but then make it so that I don't need the plotting, making it easier to be called by the API